File: Daily/PY/Leetcode740-删除并获得点数.py
class Solution:
    def deleteAndEarn(self, nums: List[int]) -> int:
        maxV= nums[0]
        for num in nums:
            maxV = max(num,maxV)
        Sum = [0] * (maxV + 1)
        for num in nums:
            Sum[num] += num
        old,now = Sum[0],Sum[1]
        n = len(Sum)
        for i in range(3,n + 1):
            tmp = max(old + Sum[i - 1],now)
            old = now
            now = tmp
        return now

File: Daily/Rust/Leetcode740-删除并获得点数.rs
impl Solution {
    pub fn delete_and_earn(nums: Vec<i32>) -> i32 {
        let &max = nums.iter().max().unwrap();
        let mut sum = vec![0;max as usize+ 1];
        for num in nums{
            sum[num as usize] += num;
        }
        let (mut old,mut now) = (sum[0],sum[1]);
        for i in 3..=max + 1{
            let tmp = now.max(old + sum[i as usize - 1]);
            old = now;
            now = tmp;
        }
        now
    }   
}